scripts/v4-sdk-docs-gen.sh

```diff
@@ -0,0 +1,125 @@
+if [ ! -d "$SDKS_SUBMODULE_DIR" ]; then
+  echo "==> Adding submodule for sdks..."
+  git submodule add --force "$SDKS_REPO_URL" "$SDKS_SUBMODULE_DIR"
+fi
+
+echo "==> Updating sdks submodule..."
+git submodule update --init --recursive
+(
+  cd "$SDKS_SUBMODULE_DIR"
+  # Adjust branch/commit if you need a specific version
+  git checkout main
+  git pull origin main
+)
+
+##############################################################################
+# 2. Install docgen dependencies (typedoc, typedoc-plugin-markdown, TS) via Yarn
+##############################################################################
+echo "==> Installing typedoc, plugin & TS in v4-sdk..."
+(
+  cd "$V4_SDK_DIR"
+  yarn add -D typedoc typedoc-plugin-markdown typescript
+  yarn install --frozen-lockfile
+)
+
+##############################################################################
+# 3. Generate Markdown docs with TypeDoc (using --skipErrorChecking)
+##############################################################################
+echo "==> Generating Markdown docs for v4-sdk..."
+(
+  cd "$V4_SDK_DIR"
+  yarn typedoc \
+    --out docs \
+    --plugin typedoc-plugin-markdown \
+    --skipErrorChecking \
+    --entryPoints src/index.ts \
+    --tsconfig tsconfig.json
+)
+
+##############################################################################
+# 4. Remove old v4 folder & copy new docs into docs/sdk/v4/reference
+##############################################################################
+echo "==> Removing old v4 folder at $DOCS_V4_ROOT..."
+rm -rf "$DOCS_V4_ROOT"
+mkdir -p "$DOCS_REFERENCE_DIR"
+
+echo "==> Copying newly generated docs to $DOCS_REFERENCE_DIR..."
+cp -R "$TYPEDOC_OUTPUT_DIR/"* "$DOCS_REFERENCE_DIR/"
+
+##############################################################################
+# 5. Rename index.md -> overview.md & remove any guides folder
+##############################################################################
+if [ -f "$DOCS_REFERENCE_DIR/index.md" ]; then
+  mv "$DOCS_REFERENCE_DIR/index.md" "$DOCS_REFERENCE_DIR/overview.md"
+fi
+if [ -f "$DOCS_REFERENCE_DIR/README.md" ]; then
+  mv "$DOCS_REFERENCE_DIR/README.md" "$DOCS_REFERENCE_DIR/overview.md"
+fi
+
+if [ -d "$DOCS_REFERENCE_DIR/guides" ]; then
+  rm -rf "$DOCS_REFERENCE_DIR/guides"
+fi
+
+##############################################################################
+# 6. Create _category_.json files (overwriting) as requested
+##############################################################################
+echo "==> Creating _category_.json in sdk/v4/ and sdk/v4/reference..."
+
+# 6a. sdk/v4/_category_.json
+mkdir -p "$DOCS_V4_ROOT"
+cat <<EOF > "$DOCS_V4_ROOT/_category_.json"
+{
+  "label": "V4 SDK",
+  "position": 1,
+  "collapsed": false
+}
+EOF
+
+# 6b. sdk/v4/reference/_category_.json
+cat <<EOF > "$DOCS_REFERENCE_DIR/_category_.json"
+{
+  "label": "Technical Reference",
+  "position": 3,
+  "collapsed": true
+}
+EOF
```

Comment:
i dont think well need this in the future; `_category_.json` are historically manually created and updated

dont want the script to overwrite manual changes - so we can discard this section